File: C#/TddApi/NavigationApp/s13.pass_second_test/Navigator.cs
using System;

namespace NavigationApp.s13.pass_second_test
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;
        Location destination;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, destination);
        }

        public void setDestination(Location dest)
        {
            this.destination = dest;
        }
    }
}

File: Python/s17_fill_controller/navigator.py
class Navigator:
    def __init__(self, starting_point, dist_provider):
        self.dist_provider = dist_provider
        self.starting_point = starting_point
        self.destination = None

    def get_distance_from_destination(self):
        return self.dist_provider.get_distance(self.starting_point, self.destination)

    def set_destination(self, dest):
        self.destination = dest
